### PR TITLE
accept patterns in allow_origins config array

### PR DESCRIPTION
**Pull request by ngabor84**
Changes proposed in this pull request:
- Accept patterns in allow_origins config array

### CODE REVIEW

**Comment by Jalle19** [comment]
The way `str_is` works I assume `*.foobar.com` would also match `a.b.foobar.com`, which may not be the intent?

File: composer.json [hunk 1]
```diff
@@ -10,7 +10,8 @@
         }
     ],
     "require": {
-        "php": ">=7.1"
+        "php": ">=7.1",
+        "illuminate/support": "^5.7"
```
Comment (Jalle19) [inline composer.json]: This should be `^5.4`, the library is supposed to work with Lumen >= 5.4

File: tests/unit/CorsServiceTest.php [hunk 2]
```diff
@@ -272,4 +272,44 @@ public function testIsPreflightRequest()
...
+    {
+        $this->request = new Request;
+
+        $this->response = new Response;
+
+        $this->service = new CorsService([
+            'allow_origins' => ['http://*.foo.com', 'http://notbar.com'],
+        ]);
+
+        $this->specify('response origin header is set when origin is match to a pattern', function () {
```
Comment (Jalle19) [inline tests/unit/CorsServiceTest.php]: The description doesn't match the test case, should probably be `when origin is not a match to a pattern`

**Review by Jalle19 — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (ngabor84) [reply composer.json]: I changed it to ^5.4

File: tests/unit/CorsServiceTest.php (hunk 2 above)
Comment (ngabor84) [reply tests/unit/CorsServiceTest.php]: Fixed

**Comment by ngabor84** [comment]
I think it's not a problem that a.b.foobar.com match pattern *.foobar.com, because in that case if I want a more strict rule I add *.b.foobar.com, but if I want to allow all of my subdomain I can do that way.

**Comment by Jalle19** [comment]
https://packagist.org/packages/nordsoftware/lumen-cors#3.1.1

Thanks for your contribution!

**Pull request merged**